packages/server/src/process_move_service.ts

```diff
@@ -0,0 +1,97 @@
+import { Board, Marker, Winner } from "@mapistry/take-home-challenge-shared";
+import { WinningLines } from "./constants";
+
+export class ProcessMoveService {
+    board: Board;
+
+    constructor(board: Board) {
+        this.board = board;
+    }
+
+    static calculateMinimaxScore(board: Board, depth: number, aiTurn: boolean) {
+        const scoreMap = {
+            O: 10,
+            X: -10,
+            tie: 0
+        }
+
+        let finalScore!: number;
+
+        const winner = this.checkForWinner(board)
+        if (winner !== null) return scoreMap[winner]
+        if (this.checkForDraw(board)) return scoreMap.tie
+
+        for (let move = 0; move < board.length; move += 1) {
+            if (board[move] === null) {
+                const simulatedBoardState = board.slice()
+                let simulatedMoveScore: number
+                if (aiTurn) {
+                    simulatedBoardState[move] = Marker.o
+                    simulatedMoveScore = ProcessMoveService.calculateMinimaxScore(simulatedBoardState, depth + 1, false) - depth
+                    if (finalScore === undefined || simulatedMoveScore > finalScore) finalScore = simulatedMoveScore
+                } else {
+                    simulatedBoardState[move] = Marker.x
+                    simulatedMoveScore = ProcessMoveService.calculateMinimaxScore(simulatedBoardState, depth + 1, true) + depth
+                    if (finalScore === undefined || simulatedMoveScore < finalScore) finalScore = simulatedMoveScore
+                }
+            }
+        }
+
+        return finalScore
+    }
+
+    static calculateScoreForMove(board: Board, move: number) {
+        const simulatedBoardState = board.slice();
+        simulatedBoardState[move] = Marker.o
+        return ProcessMoveService.calculateMinimaxScore(simulatedBoardState, 0, false)
+    }
+
+    static checkForDraw(board: Board) {
+        return board.every(square => square !== null)
+    }
+
+    static checkForWinner(board: Board) {
```

Comment:
I took the intuitive approach here and didn't put much effort into optimizing performance. Since it's such a small array, I think the approach is adequate, but if it were a larger grid or we had an emphasis on performance, we'd want to look at this.